README.md
<!-- PROJECT LOGO -->
<br />
<p align="center">
  <a href="https://github.com/axonepro/sdk-ooti/blob/master/README.md">
    <img src="ooti/logo.png" alt="Logo" width="240" height="140">
  </a>

  <h3 align="center">SDK OOTI</h3>

  <p align="center">
    <a href="https://github.com/axonepro/sdk-ooti/blob/master/README.md"><strong>Explore the docs »</strong></a>
    <br />
    <a href="https://ooti.co/">ooti.co</a>
  </p>
</p>


<!-- TABLE OF CONTENTS -->
<details open="open">
  <summary>Table of Contents</summary>
  <ol>
    <li>
      <a href="#about-the-project">About The Project</a>
      <ul>
        <li><a href="#built-with">Built With</a></li>
      </ul>
    </li>
    <li>
      <a href="#getting-started">Getting Started</a>
      <ul>
        <li><a href="#prerequisites">Prerequisites</a></li>
        <li><a href="#installation">Installation</a></li>
      </ul>
    </li>
    <li>
      <a href="#how-to-use">How to use</a>
    </li>
    <li><a href="#license">License</a></li>
  </ol>
</details>



<!-- ABOUT THE PROJECT -->
## About The Project

This sdk is for all people wanting to use the ooti API


### Built With

* [Python](https://www.python.org/)


<!-- GETTING STARTED -->
## Getting Started

### Prerequisites

* Pip 21.2.4
  - https://pip.pypa.io/en/stable/getting-started/
* Python 3.7
  - https://www.python.org/downloads/release/python-377/


### Installation
In case you work with virtuel environment you could, add ooti-api in your requirements.

Run  the following command to install package ```ooti-api```:
```sh
   pip install ooti-api
```
Installation finished, use ```make```, this create our .env, write your credential in.
```sh
    make
```

<!-- HOW TO USE -->
## How to use

1. Import
   ```py
   import sdk_ooti
   ```
2. Connect to the API
   ```py
   my_account = sdk_ooti.OotiAPI('email', 'password')
   my_account.connect()
   ```
3. Do a request
   ```py
   invoices = my_account.Invoicing.get_invoices_list()
   ``` 

## Example

```
import ooti
import pprint


###### Connect ######
my_account = ooti.OotiAPI('email', 'password')
my_account.connect()


###### Invoice ######
invoices = my_account.get_invoices_list()  # Get the list of invoices
for invoice in invoices:
    pprint.pp(invoice)  # Print each invoice

invoice = my_account.get_invoice_details(pk='invoice_pk')  # Get the details of one invoice
for line in invoice['items']:
    pprint.pp(line)  # Print each lines of the invoice

payload = {
    "amount": 1000,
}
invoice = my_account.create_invoice(1499, payload)  # Create an invoice
pprint.pp(invoice)

payload = {
    "amount": 3000,
}
invoice = my_account.update_invoice(payload, pk='invoice_pk')  # updatee an invoice
pprint.pp(invoice)

###### Payment ######
payments = my_account.get_payments_list()  # Get the list of payments
for payment in payments:
    pprint.pp(payment)  # Print each payment

payment = my_account.get_payment_details(pk='payment_pk')  # Get the details of one payment
pprint.pp(payment)

payload = {
    "amount": 1000,
}
payment = my_account.create_payment(1499, payload)  # Create an payment
pprint.pp(payment)

payload = {
    "amount": 3000,
}
payment = my_account.update_payment(payload, pk='payment_pk')  # updatee an payment
pprint.pp(payment)

###### Project ######
projects = my_account.get_projects_list()  # Get the list of projects
for project in projects:
    pprint.pp(project)  # Print each project

project = my_account.get_project_details(pk='invoice_pk')  # Get the details of one project
pprint.pp(project)  # Print details of the project

data = {
    'business_title': 'new_title'
}
project = my_account.update_project_details(pk='invoice_pk', data=data)  # Update the details of one project
pprint.pp(project)  # Print details of the updated project


###### Phase ######
phases = my_account.get_phases_list(project_pk='invoice_pk')  # Get the list of phases
for phase in phases:
    pprint.pp(phase)  # Print each phase

phase = my_account.get_phase_details(pk='phase_pk')  # Get the details of one phase
pprint.pp(phase)  # Print details of the phase

data = {'progress': 66}
phase = my_account.update_phase_details(pk='phase_pk', data=data)  # Update the details of one phase
pprint.pp(phase)  # Print details of the updated phase
```

<!-- LICENSE -->
## License

Copyright (C) 2016-2021 AxonePro S.A.S. legal@ooti.co All Rights Reserved

This file is part of axonepro.com.

Unauthorized copying of any file in this project, via any medium is strictly prohibited. Proprietary and confidential

axonepro.com can not be copied and/or distributed without the express permission of AXONEPRO.


CHANGELOG.md
Changelog
=========


(unreleased)
------------

New
- Run only one test method or class (v1.0.1) [ylesueur]
- Get_selected_org and get_selected_team instead of Factories.
  [ylesueur]
- Change log file (0.0.4) [ylesueur]
- Config file for gitchangelog API. [ylesueur]
- Makefile to init envrionement and run tests (0.0.4) [ylesueur]
- Requirements folder with dev and prod files. [ylesueur]

Changes
- Exclude .env file from deploy. [ylesueur]
- Update Changelog 22/09/2021 (v1.0.1) [ylesueur]
- Versionning (1.0.1) [Jordan martin]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Use test_helper methods instead of TeamFactory or OrgUserPK.
  [ylesueur]
- Move TeamFactory and OrguserPkFactory into test_helper. [ylesueur]
- Remove pytest warning with the attribute __test__=False (0.0.4)
  [ylesueur]
- Need to fix create_invoice_item (0.0.4) [ylesueur]
- Python for first test (0.0.4) [ylesueur]
- Requirements folder  (0.0.4) [ylesueur]
- Remove dead links from table index. [ylesueur]
- Rules ordering (0.0.4) [ylesueur]
- Move requirements.txt into requirements folder. [ylesueur]
- Get the selected organisation instead of the first one. [ylesueur]
- Add flake8, autopep8 in dev dependancies. [ylesueur]

Fix
- Folder name. [ylesueur]
- Add a clean object Invoice to create_invoice_item and
  delete_invoice_item. [ylesueur]
- __get_selected_org should be called before __get_teams to avoid None
  pk. [ylesueur]
  variable (0.0.4) [ylesueur]

Other
- Version 1.0.2. [ylesueur]
- Feat(changelog): Add changelog. [Jordan martin]
- Feat(folders-name): Change folder name, little fix in makefile.
  [Jordan martin]
- Merge pull request #31 from axonepro/0.0.4. [Max Riahi]

  Fix tests
- Fix tests. [Baptiste Vincent]
- Merge pull request #28 from axonepro/0.0.4. [Max Riahi]

  0.0.4
- Fix HelperHelper error. [Baptiste Vincent]
- Fix test_helper error. [Baptiste Vincent]
- Fix merge conflicts. [Baptiste Vincent]
- Merge pull request #26 from axonepro/0.0.3. [Max Riahi]

  0.0.4
- Merge pull request #25 from ylesueur-ooti/0.0.4. [Max Riahi]

  Fixing test and move into tests folder
- Merge branch 'master' into 0.0.4. [Baptiste Vincent]
- Merge pull request #23 from axonepro/0.0.3. [Max Riahi]

  0.0.3
- Fix for deploy 1.0.0. [Baptiste VINCENT]
- Merge pull request #27 from axonepro/structuration. [Baptiste Vincent]

  Structuration
- Fix tests. [Baptiste VINCENT]
- Fix tests. [Baptiste VINCENT]
- Remove warning test_helper. [Baptiste VINCENT]
- Fix tests. [Baptiste VINCENT]
- Fix README. [Baptiste VINCENT]
- Fix ci. [Baptiste VINCENT]
- Add teardown() functions + fix tests. [Baptiste VINCENT]
- Fill more resources + add all new tests to travis and makefile.
  [Baptiste VINCENT]
- Separate tests depending on the associated resource. [Baptiste
  VINCENT]
- Fix tests + Add test_time to travis and makefile. [Baptiste VINCENT]
- Fix tests. [Baptiste VINCENT]
- Fill last resources + delete ooti folder. [Baptiste VINCENT]
- Fill more resources. [Baptiste VINCENT]
- Move logo + remove example.py file + put example in README. [Baptiste
  VINCENT]
- Fill more resources. [Baptiste VINCENT]
- Add parameter argument for process_request method. [Baptiste VINCENT]
- Fill more resources + add process_request method + replace all
  requests. [Baptiste VINCENT]
- All requests replaced by call of process_request method. [Baptiste
  VINCENT]
- Fill costs, employees, expenses and jobs. [Baptiste VINCENT]
- Add all python files. [Baptiste VINCENT]
- Fix travis. [Baptiste VINCENT]
- Try to fix tests. [Baptiste Vincent]
- Removed __init__.py for pytest. [ylesueur]
- Fixed parent package error on import ooti module. [ylesueur]
- Contrib.md is a guideline to contributors of the sdk. [ylesueur]
- Change Log. [ylesueur]
- Added a Developer section. [ylesueur]
- Changelog. [ylesueur]
- Fix seleted orguser pk and set trips_enabled to True before testing.
  [ylesueur]
- Ignore flake8 and setup.cfg file. [ylesueur]
- Ignore flake8 and setup.cfg file. [ylesueur]
- Moved test into tests folder. [ylesueur]
- Tests moved into tests folder. [ylesueur]
- Added Flake8. [ylesueur]
- Ignore .ENV files. [ylesueur]
- Remove FreelancerTest which does not exist anymore. [ylesueur]
- Add a class to get orguser_pk. [ylesueur]
- Exclude test files from package. [ylesueur]
- Following import convention. [ylesueur]
- Fix test with TeamFactory as team selected. [ylesueur]
- Fix format string on second parameters: 1 instead of 0. [ylesueur]
- Fix test: selected right team and org. [ylesueur]
- Removed useless team selected -> factories/TeamFactory does same.
  [ylesueur]
- Move factories into new package. [ylesueur]
- Move test into separate folder. [ylesueur]
- New Folder for tests. [ylesueur]
- Get selected user team and merge with selected org method. [ylesueur]
- Version 0.0.4. [yoles]
- Add new private method to get the user selected organization. [yoles]
- Merge pull request #22 from axonepro/fixing-tests-sdk. [Max Riahi]

  Fixing tests
- Fixing tests. [Thomas]
- :memo: commented uncleaned app. [vinsJ]
- :hammer: fix auth on factories. [vinsJ]
- :hammer: fix auth with ENV variables. [vinsJ]
- Merge branch '0.0.3' of https://github.com/axonepro/sdk-ooti into
  0.0.3. [vinsJ]
- Merge pull request #21 from axonepro/pagination. [Max Riahi]

  Add Pagination
- Merge branch '0.0.3' into pagination. [Max Riahi]
- :memo: add dotenv in requirements. [vinsJ]
- :memo: remove comments. [vinsJ]
- :memo: remove comments and print. [vinsJ]
- :rocket: feat_pagination. [vinsJ]
- Merge branch '0.0.3' into pagination. [vinsJ]
- :construction: WIP - Add pagination. [vinsJ]

example.py
import sdk_ooti
import pprint


###### Connect ######
my_account = sdk_ooti.Auth('email', 'password')
my_account.connect()


###### Invoice ######
invoices = my_account.get_invoices_list()  # Get the list of invoices
for invoice in invoices:
    pprint.pp(invoice)  # Print each invoice

invoice = my_account.get_invoice_details(pk='invoice_pk')  # Get the details of one invoice
for line in invoice['items']:
    pprint.pp(line)  # Print each lines of the invoice

payload = {
    "amount": 1000,
}
invoice = my_account.create_invoice(1499, payload)  # Create an invoice
pprint.pp(invoice)

payload = {
    "amount": 3000,
}
invoice = my_account.update_invoice(pk='invoice_pk', payload)  # updatee an invoice
pprint.pp(invoice)

###### Project ######
projects = my_account.get_projects_list()  # Get the list of projects
for project in projects:
    pprint.pp(project)  # Print each project

project = my_account.get_project_details(pk='invoice_pk')  # Get the details of one project
pprint.pp(project)  # Print details of the project

data = {
    'business_title': 'new_title'
}
project = my_account.update_project_details(pk='invoice_pk', data=data)  # Update the details of one project
pprint.pp(project)  # Print details of the updated project


###### Phase ######
phases = my_account.get_phases_list(project_pk='invoice_pk')  # Get the list of phases
for phase in phases:
    pprint.pp(phase)  # Print each phase

phase = my_account.get_phase_details(pk='phase_pk')  # Get the details of one phase
pprint.pp(phase)  # Print details of the phase

data = {'progress': 66}
phase = my_account.update_phase_details(pk='phase_pk', data=data)  # Update the details of one phase
pprint.pp(phase)  # Print details of the updated phase
